notification-rule-config-service-impl/src/main/java/org/hypertrace/notification/config/service/NotificationRuleFilteredStore.java

```diff
@@ -61,14 +61,21 @@ protected Optional<NotificationRule> filterConfigData(
     return Optional.of(data)
         .filter(
             notificationRule ->
-                (!notificationRule.getNotificationRuleMutableData().getDisabled()
-                        || !filter.getExcludeDisabled())
-                    && (filter.getEventConditionTypeList().isEmpty()
-                        || filter
-                            .getEventConditionTypeList()
-                            .contains(
-                                notificationRule
-                                    .getNotificationRuleMutableData()
-                                    .getEventConditionType())));
+                satisfiesEnabledPredicate(notificationRule, filter)
```

Comment:
actually now that I see it written out, if you want you could even write it as 
```java
 return Optional.of(data)
        .filter(rule -> satisfiesEnabledPredicate(rule, filter))
        .filter(rule -> satisfiesConditionTypePredicate(rule, filter)));
```

Not strictly required, It's all about readability 😉